src/deepsparse/server/build_logger.py

```diff
@@ -0,0 +1,157 @@
+
+    :param server_config: the Server configuration model
+    :return: a DeepSparse logger instance
+    """
+
+    loggers_config = server_config.loggers
+    if loggers_config is None:
+        return loggers_config
+
+    leaf_loggers = build_leaf_loggers(loggers_config)
+    loggers = []
+
+    for name, leaf_logger in leaf_loggers.items():
+        target_logging_configs = _get_target_logging_configs(name, server_config)
+
+        if target_logging_configs:
+            logger = logger_objects.FunctionLogger(
+                logger=leaf_logger, target_logging_configs=target_logging_configs
+            )
+        else:
+            logger = leaf_logger
+
+        loggers.append(logger)
+
+    return logger_objects.MultiLogger(loggers) if len(loggers) > 1 else loggers[0]
+
+
+def build_leaf_loggers(
+    loggers_config: List[Union[str, Dict[str, Dict[str, Any]]]]
+) -> Dict[str, logger_objects.BaseLogger]:
+    """
+    Instantiate a set of leaf loggers according to the configuration
+
+    :param loggers_config: Data structure, that specifies the leaf loggers to
+        be instantiated
+    :return: A dictionary that contains a mapping from a logger's name to its instance
+    """
+
+    loggers = {}
+    for logger_config in loggers_config:
+        if isinstance(logger_config, str):
+            # instantiating a logger without arguments
+            logger_name = logger_config
+            leaf_logger = _build_single_logger(logger_name=logger_name)
+        else:
+            # instantiating a logger with arguments
+            logger_name, logger_arguments = tuple(logger_config.items())[
+                0
+            ]  # check if could be more elegant
+            leaf_logger = _build_single_logger(
+                logger_name=logger_name, logger_arguments=logger_arguments
+            )
+        loggers.update(leaf_logger)
+    return loggers
+
+
+def _get_target_logging_configs(
+    leaf_logger_name: str, server_config: ServerConfig
+) -> List[TargetLoggingConfig]:
+    list_target_logging_cfgs = []
+    # iterate over all endpoints
+    for endpoint in server_config.endpoints:
+        if endpoint.data_logging is None:
+            continue
+        # if endpoint has data logging information,
+        # iterate over endpoint's target logging config
+        for target_logging_cfg in endpoint.data_logging:
+            # new target name is a composition of the endpoint name and target name
+            target_identifier = f"{endpoint.name}.{target_logging_cfg.target}"
+            # get metric logging config corresponding to the leaf logger
+            metric_function_configs = _get_metric_function_configs(
+                leaf_logger_name, target_logging_cfg.functions
+            )
+            # append to the result
+            list_target_logging_cfgs.append(
+                TargetLoggingConfig(
+                    target=target_identifier, functions=metric_function_configs
+                )
```

Comment:
again, functionality here would be simplified and flow a bit clearer with the target moved into the metric schema